src/apiProxy/api-proxy.js

```diff
@@ -249,27 +267,11 @@ const apiProxy$ = ({ API_TIMEOUT=5000, baseUrl, duotoneUrls }) => {
 		return Rx.Observable.from(queries)    // create stream of query objects - fan-out
 			.map(queryToApiConfig)              // convert query to API-specific config
 			.map(apiConfigToRequestOptions)     // API-specific args for api request
-			.do(externalRequestOpts => request.log(['api'], JSON.stringify(externalRequestOpts.url)))  // logging
-			.concatMap(externalRequestOpts =>  // make the API calls - keep order
-				externalRequest$(externalRequestOpts)
-					.timeout(API_TIMEOUT, new Error('API response timeout'))
-					.catch(error =>
-						Rx.Observable.of(
-							[null, JSON.stringify({ error: error.message})]
-						)
-					)
-			)
-			.do(([response]) => request.log(['api'], `${response.elapsedTime}ms - ${response.request.uri.path}`)) // log api response time
-			.map(([response, body]) => body)    // ignore Response object, just process body string
-			.map(parseApiResponse)              // parse into plain object
-			.catch(error =>  // if there has been an error, all 'responses' need to be the error
-				Rx.Observable.of({ error: error.message })
-					.repeat(queries.length)
-			)
+			.do(({ url }) => request.log(['api'], JSON.stringify(url)))  // logging
 			.zip(Rx.Observable.from(queries))   // zip the apiResponse with corresponding query
-			.map(apiResponseToQueryResponse)    // convert apiResponse to app-ready queryResponse
-			.map(setApiResponseDuotones)        // special duotone prop
-			.toArray();                         // group all responses into a single array - fan-in
+			.flatMap(makeApiRequest(request, API_TIMEOUT, duotoneUrls))  // parallel requests
```

Comment:
yeah i saw that line and initially thought it might be what i wanted, but basically mergeMap with higher concurrency results in output where you can't guarantee the order of whatever is running concurrently